Adding cleanup script for specific jobs

paigerube14 opened the pull request:
Resolves OCPQE-9127

Adding cleanup script for CI usage. Takes in the workload type to be able to delete namespace that were created during the kube-burner run. Also have the option to delete benchmark-operator namespace and objects if they exist

Example: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/paige-e2e-multibranch/job/cleanup/69/console

paigerube14 commented:
Resolves OCPQE-9127

qiliRedHat commented on Jenkinsfile:
```diff
@@ -0,0 +1,119 @@
+
+  parameters {
+        string(name: 'BUILD_NUMBER', defaultValue: '', description: 'Build number of job that has installed the cluster.')
+        string(name: 'JOB', defaultValue: '', description: 'Type of job you ran and want to cleanup.')
```
Suggest to change 'JOB' to 'CI_TYPE'(same naming in loaded-upgrade job) or 'JOB_TYPE' and give an example in description. like 'e.g. node-density'

paigerube14 commented:
@SachinNinganure you should be able to run it, but yes it does appear that the benchmark-operator and namespaces all get cleaned up for that test already without needing cleanup.

SachinNinganure commented:
Thanks @paigerube14   So, this will only delete if the name of the namespace is benchmark-operator and not the namespaces that get created for instance _"acb2ca2c-12c6-487c-81e7-5926799e8be0"_ in  pod-density tests. https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/e2e-benchmarking-multibranch-pipeline/job/kube-burner/238/   
?

paigerube14 commented:
@SachinNinganure for pod-density and other kube-burner tests it will delete all the namespaces like "acb2ca2c-12c6-487c-81e7-5926799e8be0" that get created during test execution. I delete these by finding the namespaces that have label "kube-burner-job=<job-type>". The UNINSTALL_BENCHMARK_OP variable will delete the benchmark-operator and all the objects in benchmark-operator namespace if it's true

SachinNinganure commented:
And it did not delete 'acb2ca2c-12c6-487c-81e7-5926799e8be0'  namespace yet;  test run https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/paige-e2e-multibranch/job/cleanup/76/console;
cluster -->https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Flexy-install/99730/
pod-density test -->https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/e2e-benchmarking-multibranch-pipeline/job/kube-burner/238/


[sninganu@sninganu ~]$ date;oc get ns|egrep -v openshift
Thu May  5 21:24:42 IST 2022
NAME                                               STATUS   AGE
**acb2ca2c-12c6-487c-81e7-5926799e8be0**               Active   4h42m
default                                                                             Active   9h
kube-node-lease                                                             Active   9h
kube-public                                                                     Active   9h
kube-system                                                                   Active   9h

paigerube14 commented:
@SachinNinganure sorry missed a change when I renamed the top level variable. Changed it now and running: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/paige-e2e-multibranch/job/cleanup/77/

qiliRedHat commented:
@paigerube14 'router-perf cleanup both the namespaces and benchmark-operator at the end of their runs', I got you, this is designed to cover the resources NOT get deleted by the benchmark-operator. Sorry for the misunderstanding.

qiliRedHat commented:
For router-perf, sometimes the test fails in the middle, causing the resource deletion steps not get executed, so there are some resource left not deleted. But when re-run, the new test can delete the old resources if they exist before creating new, so there should be no problem.

qiliRedHat commented:
/lgtm
